tools/deploy-cli/deploy-cli.go

```diff
@@ -0,0 +1,324 @@
+	if exists && value != "" {
+		return value
+	}
+
+	return defaultValue
+}
+
+// GenerateHtpasswd generates a htpasswd entry for the given username and password.
+func GenerateHtpasswd(username, password string) (string, error) {
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("%s:%s", username, string(hashedPassword)), nil
+}
+
+// GenerateRandomString generates random string of given length
+// using crypto/rand and base64 encoding.
+func GenerateRandomString(length int) (string, error) {
+	b := make([]byte, length)
+	_, err := rand.Read(b)
+	if err != nil {
+		return "", fmt.Errorf("failed to generate random string: %v", err)
+	}
+
+	return base64.RawURLEncoding.EncodeToString(b)[:length], nil
+}
+
+// DownloadFile downloads a file and stores its content to a specified location on disk
+func DownloadFile(url string, filepath string) error {
+	// Create the file
+	out, err := os.Create(filepath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	// Get the data
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	// Write the body to file
+	_, err = io.Copy(out, resp.Body)
+	return err
+}
+
+// EnsureFileExists checks if a file exists at path, creating it with random content if it doesn't.
+func EnsureFileExists(varName, path string, length int) (bool, error) {
+	_, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		generatedString, err := GenerateRandomString(length)
+		if err != nil {
+			return false, err
+		}
+
+		err = os.WriteFile(path, []byte(generatedString), 0600)
+		if err != nil {
+			return false, err
+		}
+		log.Printf("[%s] Created new file with random content at: %s", varName, path)
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// ReadFileContent reads and returns the content of the file at path.
+func ReadFileContent(varName, path string) (string, error) {
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return "", fmt.Errorf("[%s] failed to read file %s: %v", varName, path, err)
+	}
+
+	return string(content), nil
+}
+
+// getEnvOrFileContent checks for an environment variable; if not present, ensures a file exists and reads it.
+func getEnvOrFileContent(varName, filePath string, length int) (string, error) {
+	val, exists := os.LookupEnv(varName)
+	if exists && val != "" {
+		log.Printf("[%s] Using value from environment variable", varName)
+		return val, nil
+	}
+
+	newlyCreated, err := EnsureFileExists(varName, filePath, length)
+	if err != nil {
+		return "", err
+	}
+
+	if !newlyCreated {
+		log.Printf("[%s] Reading content from existing file: %s", varName, filePath)
+	}
+
+	content, err := ReadFileContent(varName, filePath)
+	if err != nil {
+		return "", err
+	}
+
+	return content, nil
+}
+
+// BuildKustomizeManifest builds a provided kustomize overlays to output, same as `kustomize build`
+func BuildKustomizeManifest(source string) ([]byte, error) {
```

Comment:
Don't we have this in the e2e module already? Let's reuse it if we have it already and otherwise add it so that we can reuse it.

Reply:
Yes. I forgot about this, the one in e2e is currently not public, so I copied it over and made a mental note to make it public later before removing this copy.